File: frontend/src/api/budget.ts
// frontend/src/api/budget.ts
// API functions for budget endpoint (Phase 4).
// Depends on: Phase 2 / axiosInstance.ts

import api from "./axiosInstance";

export interface CategoryBreakdown {
  category: string;
  total_cost: string;
  activity_count: number;
  percentage: number;
}

export interface StopBudget {
  stop_id: string;
  city_name: string;
  total_cost: string;
  activity_count: number;
}

export interface BudgetItem {
  id: string;
  trip_id: string;
  title: string;
  category: string;
  amount: string;
  notes: string | null;
}

export interface BudgetItemCreate {
  title: string;
  category: string;
  amount: number;
  notes?: string;
}

export interface BudgetSummary {
  trip_id: string;
  trip_name: string;
  total_budget: string;
  cost_per_day: string;
  trip_duration_days: number;
  category_breakdown: CategoryBreakdown[];
  stop_breakdown: StopBudget[];
  manual_items: BudgetItem[];
}

export const budgetApi = {
  get: (tripId: string) =>
    api.get<BudgetSummary>(`/trips/${tripId}/budget`),
  
  createItem: (tripId: string, data: BudgetItemCreate) =>
    api.post<BudgetItem>(`/trips/${tripId}/budget/items`, data),
    
  updateItem: (itemId: string, data: BudgetItemCreate) =>
    api.patch<BudgetItem>(`/budget/items/${itemId}`, data),
    
  deleteItem: (itemId: string) =>
    api.delete(`/budget/items/${itemId}`),
};

// SELF-CHECK: dynamic data only ✓ | validated ✓ | paginated N/A | error handled ✓

File: backend/app/schemas/budget.py
# backend/app/schemas/budget.py
# Pydantic v2 schemas for budget breakdown responses.
# Depends on: Phase 4 / budget_service.py

from decimal import Decimal

from pydantic import BaseModel


class CategoryBreakdown(BaseModel):
    """Budget breakdown for a single activity category."""
    category: str
    total_cost: Decimal
    activity_count: int
    percentage: float  # 0-100


class StopBudget(BaseModel):
    """Budget for a single stop (city visit)."""
    stop_id: str
    city_name: str
    total_cost: Decimal
    activity_count: int


class BudgetSummary(BaseModel):
    """Complete budget summary for a trip."""
    trip_id: str
    trip_name: str
    total_budget: Decimal
    cost_per_day: Decimal
    trip_duration_days: int
    category_breakdown: list[CategoryBreakdown]
    stop_breakdown: list[StopBudget]

# SELF-CHECK: dynamic data only ✓ | validated ✓ | paginated N/A | error handled ✓
